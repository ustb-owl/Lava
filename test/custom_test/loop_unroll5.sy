void test5(int n) {
    int i = 0, j = 0, k = 0, l = 0;
    int res = 0;
    
    i = 0;
    while (i < n) {
        j = 0;
        while (j < n) {
            k = 0;
            while (k < n) {
                l = 0;
                while (l < n) {
                    res = res + i + j + k + l;
                    l = l + 1;
                }
                k = k + 1;
            }
            j = j + 1;
        }
        i = i + 1;
    }
    putint(n);
    putch(10);
    putint(i);
    putch(10);
    putint(j);
    putch(10);
    putint(k);
    putch(10);
    putint(l);
    putch(10);
    putint(res);
    putch(10);
}

int main() {
    int i;
    i = -1;
    while (i < 10) {
        i = i + 1;
        test5(i);
    }
    return 0;
}