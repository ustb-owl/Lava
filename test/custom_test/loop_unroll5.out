0
0
0
0
0
0
1
1
1
1
1
0
2
2
2
2
2
32
3
3
3
3
3
324
4
4
4
4
4
1536
5
5
5
5
5
5000
6
6
6
6
6
12960
7
7
7
7
7
28812
8
8
8
8
8
57344
9
9
9
9
9
104976
10
10
10
10
10
180000
0
